### ISSUE TITLE
Provide an option to view raw

### ISSUE THREAD

**Issue by sushruth** [markserv/markserv#5]
I do not serve my markdown files from my own machine. I set up a server instance in my company's cloud to host my markdown. I would really like an option to "download" or "view raw" of the rendered markdown I am looking at. Any chance you may implement this?

**Comment by F1LT3R** [comment]
@sushruth I am implementing this soon (next couple of weeks)

I have some of my own ideas on how this should work, but I'd love to hear how you think it should work.

I will yake your ideas into consideration as I implement HTML export.

**Comment by sushruth** [comment]
Thank you, I was also wondering if you could show "permalink" icon beside each heading. Will be easy to share links. While on the topic of permalinks for headings, would be real nice to reproduce a whole section under a heading from a different MD file. While on the topic of updating files, could be amazing to have a API like access to files and/or a command-line tool. 

(I know I am running around with a lot of possibly tedious and useless ideas but here they are anyway)

**Comment by F1LT3R** [comment]
> Thank you, I was also wondering if you could show "permalink" icon beside each heading. 

Yes, great point. Added: https://github.com/F1LT3R/markserv-cli/issues/1

> would be real nice to reproduce a whole section under a heading from a different MD file

Agreed. Added: https://github.com/F1LT3R/markserv-cli/issues/2

> could be amazing to have a API like access to files and/or a command-line tool.

Not sure what you mean. Can you give me example(s) of how you might use this?

**Comment by sushruth** [comment]
About the CLI tool, assume there is a tool which we can use like `markserv update performance.md` from a local directory, which can contact markserv to update file in the server (when server is not the same machine as what I use), maybe only when ssh-key logins are enabled, you know for security, so that we dont have to go ssh into the server and then update the file there. Am I making sense ?

**Comment by F1LT3R** [comment]
Not sure. Can you provide a little more detail?

**Comment by sushruth** [comment]
Okay, Here is a scenario : 

I am a developer working on my machine in my organization. I have a VM in my organization's cloud, which is where my markserv installation is, as of now. I have run markserv on a particular directory in that VM on a particular Port. I am using it as a pseudo-wiki for what I do, for myself. With no intention of the organization. Every single time I have a local file that needs to be updated in the VM's folder, which in fact is not on my machine, I need to SSH into the VM instance and then update that file. 

So, I was wondering, if we could have a small batch script or simple command line tool for markserv that I can install in my machine like `npm i -g markserv-updater` and then use that tool to contact that VM instance and update that file for me. like this maybe : `markserv-updater **/**.md` from a directory, probably which has a markserv.json in it to provide information about where and how to contact the markserv installation, and then update the files.

I think you can drop this if it sounds too complicated but hope this is not that complicated to implement :)

**Comment by F1LT3R** [comment]
This is interesting. How are you updating the files?

**Comment by sushruth** [comment]
currently SSH'ing into the VM and then doing a quick `rm filename.md` and then `vi filename.md` and pasting the contents. For every change. Yep. :)

**Comment by F1LT3R** [comment]
I see!

I think I could implement what you are talking about using the following package: https://github.com/serkanyersen/sync

In the meantime... what editor are you using?

**Comment by sushruth** [comment]
I use VSCODE. Oh yeah that makes sense too, to make it a plugin for editors. And yes, I could use Sync but i dont really want to monitor it always. I just need a 'as-needed-basis' thing.

**Comment by sushruth** [comment]
You know all of this would not be needed if we could just edit the file right in the browser. Maybe thats a better way ?

**Comment by F1LT3R** [comment]
Interesting idea.

https://github.com/F1LT3R/markserv-cli/issues/3

Also.. you could use something like Expandrive, though I think it costs money.

**Comment by sushruth** [comment]
Well, then I don't even have to look at it. Not gonna be allowed in a company! But I could use it personally, which I don't need to right now. Thanks, @F1LT3R !

**Comment by F1LT3R** [comment]
Np, thanks for the great ideas.

**Comment by sushruth** [comment]
Thanks for the whole thing, @F1LT3R !
